support/on-prem-archive.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+
+# The purpose of this script is to download the latest stable version of every
+# package in the core-plans repo, tar them up, and upload to S3. It also supports
+# downloading the archive from S3, extracting it, and uploading to a new depot.
+
+set -eo pipefail
+
+help() {
+    echo "Usage: on-prem-archive.sh {create-archive|populate-depot <DEPOT_URL>}"
+}
+
+exists() {
+  if command -v $1 >/dev/null 2>&1
+  then
+    return 0
+  else
+    return 1
+  fi
+}
+
+s3_cp() {
+  aws s3 cp --acl=public-read ${1} ${2} >&2
+}
+
+check_tools() {
+  for tool in "$1"
+  do
+    if ! exists "$tool"; then
+      echo "Please install $tool and run this script again."
+      exit 1
+    fi
+  done
+}
+
+check_vars() {
+  for var in "$1"
+  do
+    if [ -z "${!var}" ]; then
+      echo "Please ensure that $var is exported in your environment and run this script again."
+      exit 1
+    fi
+  done
+}
+
+bucket="habitat-on-prem-builder-bootstrap"
```

Comment:
I think top-level bucket names have to be unique, either globally or regionally. Should this be configurable?

Reply:
Yeah, when I attempt to upload, I get an error, and my S3 client tells me the name's already in use:

![image](https://user-images.githubusercontent.com/274700/36119569-21377980-0ff5-11e8-9063-4207afc82c7a.png)

Reply:
`HAB_ON_PREM_BOOTSTRAP_BUCKET_NAME` will configure this now.